src/cim/components/featureSupportLevels/ACMFeatureSupportLevelProvider.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { PropsWithChildren } from 'react';
+
+import { getOCPVersions } from '../helpers';
+import {
+  FeatureSupportLevelsMap,
+  FeatureIdToSupportLevel,
+  FeatureId,
+  SupportLevel,
+  FeatureSupportLevelContextProvider,
+  FeatureSupportLevelData,
+} from '../../../common';
+import { ClusterImageSetK8sResource } from '../../types';
+
+export type ACMFeatureSupportLevelProvider = PropsWithChildren<{
+  clusterImages: ClusterImageSetK8sResource[];
+}>;
+
+export const ACMFeatureSupportLevelProvider: React.FC<ACMFeatureSupportLevelProvider> = ({
+  children,
+  clusterImages,
+}) => {
+  const ocpVersions = React.useMemo(() => getOCPVersions(clusterImages), [clusterImages]);
+
+  const supportLevelData: FeatureSupportLevelsMap = React.useMemo<FeatureSupportLevelsMap>(() => {
+    const featureSupportLevelsMap: FeatureSupportLevelsMap = {};
+
+    ocpVersions.forEach((version) => {
+      // TODO(mlibra): get this dynamically
+      const featureIdToSupportLevel: FeatureIdToSupportLevel = {
+        REQUESTED_HOSTNAME: 'supported',
+        SNO: 'supported',
+        CNV: 'supported',
+        NETWORK_TYPE_SELECTION: 'supported',
```

Comment:
NETWORK_TYPE_SELECTION is supported always in ACM? what about in SNO?

Reply:
We will need to carefully review all features. I.e. I can see disk encryption in the ACM UI, which is not persisted. Let's do that in a separate task.